### PR TITLE
Added split info to the first child

### LINKED ISSUE #448: The first child in the `Slicer`'s chain is always "user's object"
`Slicer` always [writes](https://github.com/nspcc-dev/neofs-sdk-go/blob/2c15d978bf5a59e470669eeea3a25bfb96cdbf7f/object/slicer/slicer.go#L359) the first object in a chain with an empty split field. Such an object does not differ from a user object at all (e.g. it can be found with the `--raw` flag in `neofs-cli`).
https://github.com/nspcc-dev/neofs-sdk-go/pull/427 fixes all the other objects in a chain but not the first one. It can not be fixed with a single hotfix line, it is broken by design. Must be done carefully.

### PR DESCRIPTION
**Pull request by smallhive**
closes #448

### CODE REVIEW

File: object/slicer/slicer_test.go [hunk 1]
```diff
@@ -505,3 +506,89 @@ func (x *chainCollector) verify(in input, rootID oid.ID) {
...
+			_, err := payloadWriter.Write(payload)
+			require.NoError(t, err)
+		}
+
+		require.NoError(t, payloadWriter.Close())
+		require.Equal(t, overheadAmount+1, uint64(len(writer.headers)))
+
+		for _, h := range writer.headers {
+			splitID := h.SplitID()
+			assert.NotNil(t, splitID)
```
Comment (cthulhu-rider) [inline object/slicer/slicer_test.go:590]:
```suggestion
			require.NotNil(t, splitID)
```

**Review by cthulhu-rider — APPROVED** [review]

File: object/slicer/slicer.go [hunk 2]
```diff
@@ -355,14 +355,14 @@ func (x *PayloadWriter) Write(chunk []byte) (int, error) {
 
 	if !x.withSplit {
 		x.splitID = object.NewSplitID()
+		x.withSplit = true
```
Comment (cthulhu-rider) [inline object/slicer/slicer.go:358]: if code movement matters, and this is not obvious (`x.withSplit` isn't used in the current condition block directly, maybe inside calling methods only), i'd write a comment about this: `note: don't move, <reason>`

File: object/slicer/slicer_test.go (hunk 1 above)
Comment (smallhive) [reply object/slicer/slicer_test.go]: Updated

File: object/slicer/slicer.go (hunk 2 above)
Comment (smallhive) [reply object/slicer/slicer.go:360]: Make sense, added comment about reason

**Pull request merged**